iroha/src/lib.rs

```diff
@@ -296,6 +304,26 @@ impl Iroha {
         );
         Ok(())
     }
+
+    /// To wait for active peers(this is provisional code)
+    async fn wait_for_active_peers(&self, sorted_peers: &[PeerId]) -> Result<(), String> {
+        log::info!("Waiting for active peers");
+        for peer in sorted_peers.iter() {
+            let url = httpUri::Builder::new()
+                .scheme("http")
+                .authority(peer.address.as_str())
+                .path_and_query(uri::HEALTH_URI)
+                .build()
+                .expect(&format!("invalid peer address {}", peer.address.as_str()));
+            let response = isahc::get(url.to_string())
+                .expect(&format!("failed to send http request to {}", &url));
```

Comment:
If `isahc` fails to reach the peer, we should not fail with this error message. Instead we should retry after some time to reach the peer again, as it is a function to wait for peers.